# HG changeset patch
# Parent 9b77fdd464a1d5fec0789a1f3afb4752325f27ce
# User Mook <marky@activestate.com>
# Date 1326404636 28800
Allow (native) void* to be returned to python as integers

diff --git a/xpcom/src/VariantUtils.cpp b/xpcom/src/VariantUtils.cpp
--- a/xpcom/src/VariantUtils.cpp
+++ b/xpcom/src/VariantUtils.cpp
@@ -1738,17 +1738,21 @@ PyObject *PyXPCOM_InterfaceVariantHelper
 		break;
 	  case nsXPTType::T_CHAR:
 		ret = PyString_FromStringAndSize( ((char *)ns_v.ptr), 1 );
 		break;
 
 	  case nsXPTType::T_WCHAR:
 		ret = PyUnicode_FromPRUnichar( ((PRUnichar *)ns_v.ptr), 1 );
 		break;
-//	  case nsXPTType::T_VOID:
+	  case nsXPTType::T_VOID:
+		// we really can't do anything useful with this; just pass it to
+		// Python as an integer and hope it's close enough
+		ret = PyLong_FromVoidPtr( *((void **)ns_v.ptr));
+		break;
 	  case nsXPTType::T_IID:
 		// guard against null IIDs
 		if (*((nsIID**)ns_v.ptr) == NULL) {
 			ret = Py_None;
 			Py_INCREF(Py_None);
 		} else
 			ret = Py_nsIID::PyObjectFromIID( **((nsIID **)ns_v.ptr) );
 		break;

